compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -510,18 +593,53 @@ abstract class GenJSCode extends plugins.PluginComponent
      *  The fields are initialized with the zero of their types.
      */
     def genClassFields(cd: ClassDef): List[js.FieldDef] = {
+      val classSym = cd.symbol
+      assert(currentClassSym.get == classSym,
+          "genClassFields called with a ClassDef other than the current one")
+
       // Non-method term members are fields, except for module members.
       (for {
-        f <- currentClassSym.info.decls
+        f <- classSym.info.decls
         if !f.isMethod && f.isTerm && !f.isModule
       } yield {
         implicit val pos = f.pos
+
         val mutable =
           suspectFieldMutable(f) || unexpectedMutatedFields.contains(f)
-        js.FieldDef(encodeFieldSym(f), toIRType(f.tpe), mutable)
+        val name =
+          if (isExposed(f)) js.StringLiteral(jsNameOf(f))
+          else encodeFieldSym(f)
+
+        val irTpe = {
+          if (f.tpe.typeSymbol != CharClass || !isScalaJSDefinedJSClass(classSym))
+            toIRType(f.tpe)
+          else
+            jstpe.ClassType(ir.Definitions.BoxedCharacterClass)
+        }
```

Comment:
> I have not found a way to support this reliably so far.

What is wrong with erasing the field to `Any` in `jsinterop` and let scalac handle the rest?

In any case, please comment about this here (if you intend to not change it).